ergo-p2p/src/peer_addr.rs

```diff
@@ -0,0 +1,64 @@
+//! Peer address types
+use std::{
+    convert::TryInto,
+    net::{IpAddr, Ipv4Addr, SocketAddr, SocketAddrV4},
+};
+
+use derive_more::{From, Into};
+use sigma_ser::{ScorexSerializable, ScorexSerializationError};
+
+/// Peer address
+#[derive(PartialEq, Eq, Debug, Copy, Clone, From, Into, Hash)]
+pub struct PeerAddr(pub SocketAddr);
```

Comment:
I'd refrain from making the inner item public. It kinda goes against the idea of newtype and leaks implementation details. `.into()` should suffice for users outside of the module.